upgrade-notes/backend_20250602_110750.md

```diff
@@ -0,0 +1,5 @@
+#### improve administration right in inline edit and grid actions ([#4022](https://github.com/shopsys/shopsys/pull/4022))
+
+- revisit all your `*InlineEdit` classes, ensure that you are implementing new `getEditRole()` method and ensure that all your overridden methods are using correct security methods as in parent class
+- provide third argument `editRole` `GridFactory::create()` method to ensure, that delete action is not displayed for users without `editRole` permission
+- `Grid::addDeleteActionColumn()` now may return `null` if the user does not have permission to delete the item, so ensure that calling `setConfirmMessage()` on the returned column is safe
```

Comment:
```suggestion
provide third argument `editRole` in `GridFactory::create()` method to ensure, that delete action is not displayed for users without `editRole` permission
```

Reply:
damn, that was a mistake when providing the suggestion, I copied a wrong line :facepalm:  I should have been like this:
```suggestion
- provide third argument `editRole` in  `GridFactory::create()` method to ensure, that delete action is not displayed for users without `editRole` permission
```